optaplanner-core/src/main/java/org/optaplanner/core/api/score/constraint/ConstraintMatch.java

```diff
@@ -122,10 +125,7 @@ public boolean equals(Object o) {
 
     @Override
     public int hashCode() {
-        return (((17 * 37)
-                + constraintPackage.hashCode()) * 37
-                + constraintName.hashCode()) * 37
-                + justificationList.hashCode();
+        return Objects.hash(constraintPackage, constraintName, justificationList);
```

Comment:
+1 great idea.
Can you create a jira to look for other places that magically use the number 37 to create a hashcode. I am pretty sure I did it a lot before java 8 come in.

Reply:
https://issues.jboss.org/browse/PLANNER-1725